### PR TITLE
Remove shortened operations name and maintain operation order in blocks.

### LINKED ISSUE #2874: Remvoe OperationIds type redefinition and correct ordering
MAKE A PR TOWARDS BRANCH [release_testnet_14](https://github.com/massalabs/massa/tree/release_testnet_14)

Operation order in blocks matters:

* operation order is chosen by the block producer
* operations are executed in the order they appear in the block
* the operation global hash in the header is sensitive to operation order
* it is therefore consensus-critical for the operations order to be the same for everyone for every block

However, OperationIds seems to be used almost everywhere, even in places where operation order should matter (see https://github.com/massalabs/massa/pull/2862#discussion_r946848973 but also in Protocol etc...)

Given how little benefit and how huge of a problem this type redefinition brings, I think it would be smart to completely remove it and write explicitly `Set<OperationId>` or `Vec<OperationId>` depending on the situation. It wasn't really that much shorter anyways.

### PR DESCRIPTION
**Pull request by AurelienFT**
Fix #2874

### CODE REVIEW

**Review by adrien-zinger — APPROVED** [review]

File: massa-protocol-worker/src/checked_operations.rs [hunk 1]
```diff
@@ -21,7 +21,7 @@ impl CheckedOperations {
         self.0.insert(prefix, *id).is_none()
     }
 
-    /// Get a set of [OperationIds] matching with the givec `prefix`.
+    /// Get a set of [Set<OperationId>] matching with the givec `prefix`.
```
Comment (damip) [inline massa-protocol-worker/src/checked_operations.rs:24]: To me it looks more like the function returns an Option with just a single operation ID

File: massa-storage/src/endorsement_indexes.rs [hunk 2]
```diff
@@ -40,28 +46,47 @@ impl EndorsementIndexes {
...
+    /// - endorsement_ids: the endorsements to link to the block
     pub(crate) fn link_endorsements_with_block(
         &mut self,
         block_id: &BlockId,
-        operations: &Vec<EndorsementId>,
+        endorsement_ids: &Vec<EndorsementId>,
     ) {
         self.index_by_block
             .entry(*block_id)
             .or_default()
```
Comment (damip) [inline massa-storage/src/endorsement_indexes.rs:60]: This won't insert the element if it is missing.

Use `.or_insert_with(|| Default::default())`

File: massa-storage/src/operation_indexes.rs [hunk 3]
```diff
@@ -7,12 +10,15 @@ pub struct OperationIndexes {
     /// Operation structure container
     pub(crate) operations: Map<OperationId, WrappedOperation>,
     /// Structure mapping creators with the created operations
-    index_by_creator: Map<Address, OperationIds>,
+    index_by_creator: Map<Address, Set<OperationId>>,
     /// Structure mapping block ids with the operations
-    index_by_block: Map<BlockId, OperationIds>,
+    index_by_block: Map<BlockId, Vec<OperationId>>,
```
Comment (damip) [inline massa-storage/src/operation_indexes.rs:15]:
```suggestion
    index_by_block: Map<BlockId, Set<OperationId>>,
```

**Comment by AurelienFT** [comment]
Done in https://github.com/massalabs/massa/pull/2862/commits/30d79ddb16024dd6576145fb337b7371b9ea7f85

**Pull request merged**